src/node_buffer.cc

```diff
@@ -445,12 +445,13 @@ MaybeLocal<Object> New(Environment* env,
 namespace {
 
 void CreateFromString(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
   CHECK(args[0]->IsString());
-  CHECK(args[1]->IsString());
+  CHECK(args[1]->IsInt32());
 
-  enum encoding enc = ParseEncoding(args.GetIsolate(),
-                                    args[1].As<String>(),
-                                    UTF8);
+  enum encoding enc =
+    static_cast<enum encoding>(args[1]->Int32Value(env->context()).ToChecked());
```

Comment:
```suggestion
    static_cast<enum encoding>(args[1].As<Int32>()->Value());
```

(since you already CHECK’ed that it’s an Int32)